samples/nrf9160/modem_shell/src/location/location_srv_ext_lwm2m.c

```diff
@@ -0,0 +1,188 @@
+		 */
+		k_sleep(K_SECONDS(1));
+	}
+	if (err) {
+		mosh_error("Failed to request A-GPS data, err: %d", err);
+	}
+}
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+void location_srv_ext_pgps_handle(const struct gps_pgps_request *pgps_req)
+{
+	int err = -1;
+
+	if (!cloud_lwm2m_is_connected()) {
+		mosh_error("LwM2M not connected, can't request P-GPS data");
+		return;
+	}
+
+	err = location_assist_pgps_set_prediction_count(pgps_req->prediction_count);
+	err |= location_assist_pgps_set_prediction_interval(pgps_req->prediction_period_min);
+	location_assist_pgps_set_start_gps_day(pgps_req->gps_day);
+	err |= location_assist_pgps_set_start_time(pgps_req->gps_time_of_day);
+	if (err) {
+		mosh_error("Failed to set P-GPS request parameters");
+		return;
+	}
+
+	while ((err = location_assistance_pgps_request_send(cloud_lwm2m_client_ctx_get(), true)) ==
+	       -EAGAIN) {
+		/* LwM2M client utils library is currently handling an A-GPS data request, need to
+		 * wait until it has been completed.
+		 */
+		k_sleep(K_SECONDS(1));
+	}
+	if (err) {
+		mosh_error("Failed to request P-GPS data, err: %d", err);
+	}
+}
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+#if defined(CONFIG_LOCATION_METHOD_CELLULAR)
+static int location_ctrl_lwm2m_cell_result_cb(uint16_t obj_inst_id,
+					      uint16_t res_id, uint16_t res_inst_id,
+					      uint8_t *data, uint16_t data_len,
+					      bool last_block, size_t total_size)
+{
+	int err;
+	int32_t result;
+	struct location_data location = {0};
+	double temp_accuracy;
+
+	if (data_len != sizeof(int32_t)) {
+		mosh_error("Unexpected data length in callback");
+		err = -1;
+		goto exit;
+	}
+
+	result = (int32_t)*data;
+	switch (result) {
+	case LOCATION_ASSIST_RESULT_CODE_OK:
+		break;
+
+	case LOCATION_ASSIST_RESULT_CODE_PERMANENT_ERR:
+	case LOCATION_ASSIST_RESULT_CODE_TEMP_ERR:
+	default:
+		mosh_error("LwM2M server failed to get location");
+		err = -1;
+		goto exit;
+	}
+
+	err = lwm2m_engine_get_float(LWM2M_PATH(GROUND_FIX_OBJECT_ID, 0, GROUND_FIX_LATITUDE),
+				     &location.latitude);
+	if (err) {
+		mosh_error("Getting latitude from LwM2M engine failed, err: %d", err);
+		goto exit;
+	}
+
+	err = lwm2m_engine_get_float(LWM2M_PATH(GROUND_FIX_OBJECT_ID, 0, GROUND_FIX_LONGITUDE),
+				     &location.longitude);
+	if (err) {
+		mosh_error("Getting longitude from LwM2M engine failed, err: %d", err);
+		goto exit;
+	}
+
+	err = lwm2m_engine_get_float(LWM2M_PATH(GROUND_FIX_OBJECT_ID, 0, GROUND_FIX_ACCURACY),
+				     &temp_accuracy);
+	if (err) {
+		mosh_error("Getting accuracy from LwM2M engine failed, err: %d", err);
+		goto exit;
+	}
+	location.accuracy = temp_accuracy;
+
+exit:
+	if (err) {
+		location_cellular_ext_result_set(LOCATION_EXT_RESULT_ERROR, NULL);
+	} else {
+		location_cellular_ext_result_set(LOCATION_EXT_RESULT_SUCCESS, &location);
+	}
+
+	return err;
+}
+
+void location_srv_ext_cellular_handle(
+	const struct lte_lc_cells_info *cell_info,
+	bool cloud_resp_enabled)
```

Comment:
@tokangas Is there a way to use `cloud_resp_enabled` with lwm2m? Can we leave the callback out and indicate unknown result to Location library?

Reply:
Actually there is. I think you can just use `cloud_resp_enabled` as a parameter for `ground_fix_set_report_back()` below. Of course you would also have to call `location_cellular_ext_result_set()` here as well when no response is asked back.

Reply:
I've done some changes but I think you should verify this and likely need some changes still so it's better you do them as you have the setup.
I don't know if the callback is always called even if no response is expected. And whether `confirmable` parameter should be set to false (I thought it should still be true) for `location_assistance_ground_fix_request_send()`.

Reply:
The implementation looks good to me. I think it should be enough to just call `ground_fix_set_report_back()`, the callback can be always set. When g`round_fix_set_report_back()` is called with false, the device should not get the location back and the callback should not get called, I think. And I think `confirmable` should still be `true`. My understanding is that this is just QoS for the request. Even if we're not interested in the actual location in the device, we do want the request to reach the cloud.

I'll give it a try anyway.

Reply:
Tried this with LwM2M and works fine with and without --cellular_resp_disabled parameter.